src/transformers/models/deberta/modeling_tf_deberta.py

```diff
@@ -0,0 +1,1520 @@
+
+from ...activations_tf import get_tf_activation
+from ...file_utils import add_code_sample_docstrings, add_start_docstrings, add_start_docstrings_to_model_forward
+from ...modeling_tf_outputs import (
+    TFBaseModelOutput,
+    TFMaskedLMOutput,
+    TFQuestionAnsweringModelOutput,
+    TFSequenceClassifierOutput,
+    TFTokenClassifierOutput,
+)
+from ...modeling_tf_utils import (
+    TFMaskedLanguageModelingLoss,
+    TFModelInputType,
+    TFPreTrainedModel,
+    TFQuestionAnsweringLoss,
+    TFSequenceClassificationLoss,
+    TFTokenClassificationLoss,
+    get_initializer,
+    input_processing,
+    shape_list,
+)
+from ...utils import logging
+from .configuration_deberta import DebertaConfig
+
+
+logger = logging.get_logger(__name__)
+
+tnp.experimental_enable_numpy_behavior()
+
+
+_CONFIG_FOR_DOC = "DebertaConfig"
+_TOKENIZER_FOR_DOC = "DebertaTokenizer"
+_CHECKPOINT_FOR_DOC = "kamalkraj/deberta-base"
+
+TF_DEBERTA_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "kamalkraj/deberta-base",
+    # See all BrandNewBERT models at https://huggingface.co/models?filter=DeBERTa
+]
+
+
+class ContextPooler(tf.keras.layers.Layer):
+    def __init__(self, config: DebertaConfig, **kwargs):
+        super().__init__(**kwargs)
+        self.dense = tf.keras.layers.Dense(config.pooler_hidden_size, name="deberta_pooler_dense")
+        self.dropout = StableDropout(config.pooler_dropout, name="deberta_pooler_dropout")
+        self.config = config
+
+    def call(self, hidden_states, training: bool = False):
+        # We "pool" the model by simply taking the hidden state corresponding
+        # to the first token.
+        context_token = hidden_states[:, 0]
+        context_token = self.dropout(context_token, training=training)
+        pooled_output = self.dense(context_token)
+        pooled_output = get_tf_activation(self.config.pooler_hidden_act)(pooled_output)
+        return pooled_output
+
+    @property
+    def output_dim(self) -> int:
+        return self.config.hidden_size
+
+
+class XSoftmax(tf.keras.layers.Layer):
+    """
+    Masked Softmax which is optimized for saving memory
+
+    Args:
+        input (:obj:`torch.tensor`): The input tensor that will apply softmax.
+        mask (:obj:`torch.IntTensor`): The mask matrix where 0 indicate that element will be ignored in the softmax calculation.
+        dim (int): The dimension that will apply softmax
+    """
+
+    def __init__(self, axis=-1, **kwargs):
+        super().__init__(**kwargs)
+        self.axis = axis
+
+    def call(self, inputs: tf.Tensor, mask: tf.Tensor):
+
+        rmask = tf.logical_not(tf.cast(mask, tf.bool))
+        output = tf.where(rmask, float("-inf"), inputs)
+        output = tf.nn.softmax(output, self.axis)
+        output = tf.where(rmask, 0.0, output)
+        return output
+
+
+def get_mask(input, dropout):
+    mask = tf.cast(1 - tfp.distributions.Bernoulli(probs=1 - dropout).sample(sample_shape=shape_list(input)), tf.bool)
```

Comment:
It looks like this is the only use of `tfp`, which can easily be replaced by picking random number between 0 and 1 and comparing to `1-dropout`. So I think we can remove that extra dependency.

Reply:
yes. I can make that change. The original PyTorch implementation use Bernoulli distribution. 
or
to remove the extra dependency also I can use `tf.compat.v1.distributions.Bernoulli`

Reply:
removed `tfp`